daemon/container/view.go

```diff
@@ -294,14 +294,9 @@ func (v *View) GetAllNames() map[string][]string {
 func (v *View) transform(ctr *Container) *Snapshot {
 	health := container.NoHealthcheck
 	failingStreak := 0
-	if ctr.Health != nil {
-		health = ctr.Health.Status()
-		failingStreak = ctr.Health.FailingStreak
-	}
-
-	healthSummary := &container.HealthSummary{
-		Status:        health,
-		FailingStreak: failingStreak,
+	if ctr.State.Health != nil {
+		health = ctr.State.Health.Status()
+		failingStreak = ctr.State.Health.Health.FailingStreak
```

Comment:
For others; I pushed an extra commit that also exposes places where we access `State.Health.Health.<field>` - `State.Health` contains a mutex, but we have various places where we access fields inside it without obtaining a lock.

_Possibly_ some of those are "ok", but trying to expose the ugliness so that we can look at those uses, and where needed make design changes to make sure it's robust.